packages/captp/src/captp.js

```diff
@@ -436,18 +727,44 @@ export function makeLoopback(ourId) {
     }),
   });
 
+  const slotBody = JSON.stringify({
+    '@qclass': 'slot',
+    index: 0,
+  });
+
   // Create the tunnel.
   let farDispatch;
-  const { dispatch: nearDispatch, getBootstrap: getFarBootstrap } = makeCapTP(
-    `near-${ourId}`,
-    o => farDispatch(o),
-    bootstrap,
-  );
-  const { dispatch, getBootstrap: getNearBootstrap } = makeCapTP(
-    `far-${ourId}`,
-    nearDispatch,
-    bootstrap,
-  );
+  const {
+    Trap,
+    dispatch: nearDispatch,
+    getBootstrap: getFarBootstrap,
+  } = makeCapTP(`near-${ourId}`, o => farDispatch(o), bootstrap, {
```

Comment:
Nice trick to tolerate the delayed initialization of `farDispatch`. Given this trick, couldn't this instead be a use-before-define of a `const` declaration? Then, if you mess up, you get a TDZ error which is hopefully clearer.

Reply:
Resolving